clippy_utils/src/hir_utils.rs

```diff
@@ -83,6 +86,38 @@ impl<'a, 'tcx> SpanlessEq<'a, 'tcx> {
     pub fn eq_path_segments(&mut self, left: &[PathSegment<'_>], right: &[PathSegment<'_>]) -> bool {
         self.inter_expr().eq_path_segments(left, right)
     }
+
+    fn cannot_be_compared_block(&mut self, block: &Block<'_>) -> bool {
+        if block.stmts.first().map_or(false, |stmt| {
```

Comment:
The last statement would work better. This should be caught as well:
```rust
let foo = do_something();
todo!("do bar");
```

Reply:
I did in https://github.com/rust-lang/rust-clippy/pull/9006/commits/46d056e2eb7c5cab767c91383cf7ef1828809fd9